pkg/lib/synthesize/non_inferrable_csv.go

```diff
@@ -0,0 +1,31 @@
+package synthesize
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/api/pkg/lib/version"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+type RegistryV2CSV struct {
+	Name                      string                           `json:"name,omitempty"`
+	GenerateName              string                           `json:"generateName,omitempty"`
+	Kind                      string                           `json:"kind"`
```

Comment:
At first glance, the `Kind` discriminator made me think this was a kube type. Do we need this on a per-metadata-file basis or can we get away with versioning the bundles as a whole; i.e. `annotations.yaml`, `dependencies.yaml`, and this are all versioned as a group?

Reply:
I agree - let's not have a `Kind` here. Versioning as a group makes sense and can be handled by the mediatypes in annotations.json

Reply:
ok then, let's do that

Reply:
Since my impression of the metadata dir and all the files within can be randomly named and the metadata dir name will be specified in annotations.yaml. I am imposing that we parse the annotations.yaml from the same dir and if not file is parable to that format, we check if the dir is named metadata.

Reply:
nvm, just gonna add an input .